pkg/register/document.go

```diff
@@ -122,24 +121,24 @@ func (document RegisterDocument) Validate() []error {
 	}
 
 	// Check key/delegation names are unique in the document
-	bufOfNames := ""
+	bufOfNames := map[string]bool{}
 
 	for _, publicKey := range document.PublicKeys {
 		errs = append(errs, validatePublicKey(PublicKeyTypeString, publicKey)...)
 
-		if strings.Contains(bufOfNames, publicKey.ID+"|") {
+		if _, found := bufOfNames[publicKey.ID]; found {
 			errs = append(errs, fmt.Errorf("key name '%s' is not unique", publicKey.ID))
 		}
-		bufOfNames = bufOfNames + publicKey.ID + "|"
+		bufOfNames[publicKey.ID] = true
```

Comment:
(we could add a continue in the if to avoid updating the map if already present)

Reply:
Thanks, I've added the continues.